lib/mirage.ml

```
@@ -33,6 +33,83 @@ let driver_error name =
   Printf.sprintf "fail (Failure %S)" name
 
 (** {2 Devices} *)
+type qrexec = job
+let qrexec = job
+
+let qrexec_qubes = impl @@ object
+  inherit base_configurable
+  method ty = qrexec
+  val name = Name.ocamlify @@ "qrexec_"
+  method name = name
+  method module_name = "Qubes.RExec"
+  method packages = Key.pure ["mirage-qubes"]
+  method libraries = Key.pure ["mirage-qubes"]
+  method connect i modname _args =
+  match Key.(get (Info.context i) target) with
+  | `Unix | `MacOSX | `Xen ->
+  Fmt.strf
+  "return (`Ok ())"
```

We should probably report an error if you ask for rexec on a non-Qubes platform.